### PR TITLE
feat: insert_hugr/insert_view return node map

### PR DESCRIPTION
**Pull request by acl-cqc**
Also corresponding builder methods. `insert_hugr_internal` already returns this, so just wrap in a new `struct InsertionResult` translating the NodeIndex's into Nodes.

### CODE REVIEW

**Comment by croyzor** [comment]
I do think having an `InsertionResult` type struct would be a bit clearer

File: src/hugr/rewrite/outline_cfg.rs [hunk 1]
```diff
@@ -128,26 +129,21 @@ impl Rewrite for OutlineCfg {
...
-            new_block_bldr.set_outputs(pred_wire, cfg_outputs).unwrap();
-            h.insert_hugr(outer_cfg, new_block_bldr.hugr().clone())
-                .unwrap()
+            new_block_bldr
+                .set_outputs(pred_wire, cfg.outputs())
+                .unwrap();
+            let (new_block, node_map) = h
+                .insert_hugr(outer_cfg, new_block_bldr.hugr().clone())
+                .unwrap();
+            (new_block, *node_map.get(&cfg.node()).unwrap())
```
Comment (croyzor) [inline src/hugr/rewrite/outline_cfg.rs:143]: The second element here should just be `new_block`, no?

Comment (croyzor) [reply src/hugr/rewrite/outline_cfg.rs:143]: In which case all of the changes to return nodemaps are redundant and all we needed was to bind the result of `insert_hugr`

Comment (croyzor) [reply src/hugr/rewrite/outline_cfg.rs:143]: Ah I see that's what was happening before and the Cfg is meant to be its child

**Review by croyzor — APPROVED** [review]

**Comment by acl-cqc** [comment]
> I do think having an `InsertionResult` type struct would be a bit clearer

Fair enough! Done :)

**Pull request merged**